src/lib.rs

```diff
@@ -92,76 +126,23 @@ pub fn hash(data: &[u8]) -> [u8; 32] {
     hashing_impl::sha3_256(data)
 }
 
-/// Represents a set of public keys, consisting of a public signature key and a public
-/// encryption key.
+/// Use `gen_encrypt_keypair()` to generate a public and private key pair.
 #[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Clone)]
-pub struct PublicKeys {
-    sign: sign::PublicKey,
+pub struct PublicEncryptKey {
     encrypt: box_::PublicKey,
 }
 
-/// Secret counterpart of the public key set, consisting of a secret signing key and
-/// a secret encryption key.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct SecretKeys {
-    inner: Arc<SecretKeysInner>,
-    public: PublicKeys,
-}
-
-#[derive(Debug, PartialEq, Eq)]
-struct SecretKeysInner {
-    sign: sign::SecretKey,
-    encrypt: box_::SecretKey,
-}
-
-/// Secret key for authenticated symmetric encryption.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct SymmetricKey {
-    encrypt: Arc<secretbox::Key>,
-}
-
-/// Detached signature.
-#[derive(Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, Clone)]
-pub struct Signature {
-    signature: sign::Signature,
-}
-
-/// Precomputed shared secret key.
-/// Can be created from a pair of our secret key and the recipient's public key.
-/// As a result, we'll get the same key as the recipient with their secret key and
-/// our public key.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct SharedSecretKey {
-    precomputed: Arc<box_::PrecomputedKey>,
-}
-
-#[derive(Serialize, Deserialize)]
-struct CipherText {
-    nonce: [u8; box_::NONCEBYTES],
-    ciphertext: Vec<u8>,
-}
+impl PublicEncryptKey {
+    /// Construct public key from bytes. Useful when it was serialized before.
+    pub fn from_bytes(public_key: [u8; PUBLIC_ENCRYPT_KEY_BYTES]) -> Self {
+        Self {
+            encrypt: box_::PublicKey(public_key),
+        }
+    }
 
-impl PublicKeys {
-    /// Returns a public signing key.
-    #[cfg(feature = "mock")]
-    pub fn public_sign_key(&self) -> PublicSignKey {
-        // Pads the key to 32 bytes for mock-crypto
-        let mut full_len_key = [0; 32];
-        self.sign
-            .0
-            .iter()
-            .cloned()
-            .cycle()
-            .take(32)
-            .enumerate()
-            .for_each(|(i, v)| full_len_key[i] = v);
-        full_len_key
-    }
-
-    /// Returns a public signing key.
-    #[cfg(not(feature = "mock"))]
-    pub fn public_sign_key(&self) -> PublicSignKey {
-        self.sign.0
+    /// For anyone who wants to store the public key.
```

Comment:
Rather than a specific use-case, it might be better to just state that this converts the `PublicEncryptKey` into the raw underlying bytes or something like that?

Also, [a Rust convention](https://rust-lang-nursery.github.io/rust-clippy/v0.0.207/index.html#wrong_self_convention) is that a method like this which consume `self` should be named `into_...`.  I just noticed that [clippy is actually failing on this PR](https://travis-ci.org/maidsafe/safe_crypto/jobs/415933111) because of these `as_bytes()` methods (but it didn't show as a failure on the CI run because of a clippy bug which is resolved in a later version of clippy).

Reply:
Ah thanks, ok I've updated the names to `into_...`